fiftyone/brain/internal/core/uniqueness.py

```diff
@@ -166,20 +155,28 @@ def compute_uniqueness(
     logger.info("Uniqueness computation complete")
 
 
-def _compute_uniqueness(sample_ids, similarity_index, n_neighbors=3):
-    num_samples = len(sample_ids)
-    if num_samples <= n_neighbors:
-        return [1] * num_samples
+def _compute_uniqueness(
+    embeddings, similarity_index, batch_size=10, progress=None
+):
+    K = 3
 
-    dists_list = []
-    if similarity_index.config.method != "sklearn":
-        logger.info("This computation may take a while.")
-    for sample_id in sample_ids:
-        _, dist = similarity_index._kneighbors(
-            query=sample_id, k=n_neighbors + 1, return_dists=True
-        )
-        dists_list.append(dist)
-    dists = np.array(dists_list)
+    num_embeddings = len(embeddings)
+    if num_embeddings <= K:
+        return [1] * num_embeddings
+
+    if similarity_index.config.method == "sklearn":
+        _, dists = similarity_index._kneighbors(k=K + 1, return_dists=True)
+    else:
+        dists = []
+        with fou.ProgressBar(total=num_embeddings, progress=progress) as pb:
```

Comment:
This lays the groundwork for batching. The `_kneighbors()` interface already allows it, but none of the backends currently do anything smarter than sequential queries. However, if/when backends do support batched queries, this will automatically leverage it.

Reply:
Yeah that would currently require the person running the test to have one of the non-builtin similarity backends installed. I did manually verify that the code works by running the code in the PR description with `backend="qdrant"`.